source/opt/inline_pass.cpp

```diff
@@ -270,30 +319,120 @@ std::unique_ptr<BasicBlock> InlinePass::AddGuardBlock(
   return new_blk_ptr;
 }
 
+DebugScope InlinePass::BuildDebugScope(const DebugScope& callee_instr_scope,
+                                       DebugInlinedAtContext& inlined_at_ctx) {
+  return DebugScope(callee_instr_scope.GetLexicalScope(),
+                    BuildDebugInlinedAtChain(callee_instr_scope.GetInlinedAt(),
+                                             inlined_at_ctx));
+}
+
+uint32_t InlinePass::BuildDebugInlinedAtChain(
+    uint32_t callee_inlined_at, DebugInlinedAtContext& inlined_at_ctx) {
+  if (inlined_at_ctx.FunctionCallScope().GetLexicalScope() == kNoDebugScope)
+    return kNoInlinedAt;
+
+  // Reuse the already generated DebugInlinedAt chain if exists.
+  uint32_t already_generated_chain_head_id =
+      inlined_at_ctx.InlinedAtChainHead(callee_inlined_at);
+  if (already_generated_chain_head_id != kNoInlinedAt) {
+    return already_generated_chain_head_id;
+  }
```

Comment:
The function and variable names are not very meaningful.  I have no idea what this is looking for or in what circumstances it will happen.  `InlinedAtChainHead` sounds like it should return a bool that says whether or not something has already been inlined at the chain head.

You don't have any comments at the declaration of the function.